src/leveldbwrapper.h

```diff
@@ -85,12 +92,35 @@ class CLevelDBWrapper
     //! the database itself
     leveldb::DB* pdb;
 
+    //! a key used for optional XOR-obfuscation of the database
+    std::string obfuscate_key;
+
+    //! the key under which the obfuscation key is stored
+    static const std::string OBFUSCATE_KEY_KEY;
+    
+    //! the length of the obfuscate key in number of bytes
+    static const unsigned int OBFUSCATE_KEY_NUM_BYTES;
+    
+    std::string CreateObfuscateKey() const;
+
 public:
-    CLevelDBWrapper(const boost::filesystem::path& path, size_t nCacheSize, bool fMemory = false, bool fWipe = false);
+    /**
+     * @param[in] path        Location in the filesystem where leveldb data will be stored.
+     * @param[in] nCacheSize  Configures various leveldb cache settings.
+     * @param[in] fMemory     If true, use leveldb's memory environment.
+     * @param[in] fWipe       If true, remove all existing data.
+     * @param[in] obfuscate   If true, store data obfuscated via simple XOR. If false, XOR
+     *                        with a zero'd byte array.
+     */
+    CLevelDBWrapper(const boost::filesystem::path& path, size_t nCacheSize, bool fMemory = false, bool fWipe = false, bool obfuscate = false);
     ~CLevelDBWrapper();
 
+    /**
+     * @param[in] unobfuscate   This should always be true unless we're reading
+     *                          OBFUSCATE_KEY_KEY.
+     */
     template <typename K, typename V>
-    bool Read(const K& key, V& value) const throw(leveldb_error)
+    bool Read(const K& key, V& value, bool unobfuscate = true) const throw(leveldb_error)
```

Comment:
Is there any other case where `unobfuscate == false`, except in the [constructor](https://github.com/bitcoin/bitcoin/pull/6650/files#diff-77251d49f77d3e19d7d7ca887cd65e07R76)? If not, you might get away without it (and the condition in [L140](https://github.com/bitcoin/bitcoin/pull/6650/files#diff-0785f0fd15c21eecd3eee494a531d37fR140)), because `obfuscate_key` is still blank during the [read of the actual key](https://github.com/bitcoin/bitcoin/pull/6650/files#diff-77251d49f77d3e19d7d7ca887cd65e07R76), if I'm not mistaken.